uamqp/authentication/cbs_auth_async.py

```diff
@@ -221,3 +245,40 @@ class JWTTokenAsync(JWTTokenAuth, CBSAsyncAuthMixin):
      Default is 'UTF-8'.
     :type encoding: str
     """
+
+    def __init__(self, audience, uri,
+                 get_token,
+                 expires_in=datetime.timedelta(seconds=constants.AUTH_EXPIRATION_SECS),
+                 expires_at=None,
+                 port=constants.DEFAULT_AMQPS_PORT,
+                 timeout=10,
+                 retry_policy=TokenRetryPolicy(),
+                 verify=None,
+                 token_type=b"jwt",
+                 http_proxy=None,
+                 transport_type=TransportType.Amqp,
+                 encoding='UTF-8'):  # pylint: disable=no-member
+        self._retry_policy = retry_policy
+        self._encoding = encoding
+        self.uri = uri
+        parsed = compat.urlparse(uri)  # pylint: disable=no-member
+
+        self.cert_file = verify
+        self.hostname = parsed.hostname.encode(self._encoding)
+
+        if not is_coroutine(get_token):
+            raise ValueError("get_token must be a coroutine function")
```

Comment:
You don't need to raise a ValueError here because the `is_coroutine` function already raises it